src/d3a/d3a_core/sim_results/market_price_energy_day.py

```diff
@@ -72,40 +59,33 @@ def calculate_csv_output(self, area_dict, price_energy_redis_output):
             self.csv_output[area_dict['name']] = {
                 "price-currency": "Euros",
                 "load-unit": "kWh",
-                "price-energy-day": [],
-                "grid-fee-rate": []
+                "price-energy-day": []
             }
         self.csv_output[area_dict['name']]["price-energy-day"].\
             append(price_energy_redis_output[area_dict['uuid']]['price-energy-day'])
-        self.csv_output[area_dict['name']]["grid-fee-rate"].\
-            append(price_energy_redis_output[area_dict['uuid']]['grid-fee-rate'])
         for child in area_dict['children']:
             self.calculate_csv_output(child, price_energy_redis_output)
 
     @staticmethod
-    def _convert_output_format(price_energy, redis_output, fees):
-        for node_name, trade_rates in price_energy.items():
-            if node_name not in redis_output:
-                redis_output[node_name] = {
+    def _convert_output_format(price_energy, redis_output, core_stats):
+        for node_uuid, trade_rates in price_energy.items():
+            if node_uuid not in redis_output:
+                redis_output[node_uuid] = {
                     "price-currency": "Euros",
                     "load-unit": "kWh",
-                    "price-energy-day": [],
-                    "grid-fee-rate": []
+                    "price-energy-day": []
                 }
-            redis_output[node_name]["price-energy-day"] = [
+            redis_output[node_uuid]["price-energy-day"] = [
                 {
                     "time": timeslot,
                     "av_price": round_floats_for_ui(mean(trades) if len(trades) > 0 else 0),
                     "min_price": round_floats_for_ui(min(trades) if len(trades) > 0 else 0),
                     "max_price": round_floats_for_ui(max(trades) if len(trades) > 0 else 0),
                 } for timeslot, trades in trade_rates.items()
             ]
-        for node_name, fee_rates in fees.items():
-            redis_output[node_name]["grid-fee-rate"] = [
-                {
-                    "time": timeslot,
-                    "av_price": round_floats_for_ui(mean(fee_rate) if len(fee_rate) > 0 else 0),
-                    "min_price": round_floats_for_ui(min(fee_rate) if len(fee_rate) > 0 else 0),
-                    "max_price": round_floats_for_ui(max(fee_rate) if len(fee_rate) > 0 else 0),
-                } for timeslot, fee_rate in fee_rates.items()
-            ]
+
+            area_core_stats = core_stats.get(node_uuid, {})
+            fee = area_core_stats['grid_fee_constant'] / 100 \
+                if key_in_dict_and_not_none(area_core_stats, 'grid_fee_constant') else None
+
+            redis_output[node_uuid]["price-energy-day"][0].update({"fee_price": fee})
```

Comment:
Just for my understanding, since we always calculate one market slot at a time, I guess that there is always going to be one element in this list, right ?

Reply:
Yes!